src/dev/i18n/utils.js

```diff
@@ -125,6 +127,30 @@ export function createParserErrorMessage(content, error) {
   return `${error.message}:\n${context}`;
 }
 
+/**
+ * Recursively extracts all references from ICU message ast.
+ *
+ * @param {any} node
+ * @param {Set<string>} keys
+ */
+function extractValueReferencesFromIcuAst(node, keys = new Set()) {
```

Comment:
__note:__ please add a couple of unit tests for the change and add examples of the ICU strings we're parsing to the JSDoc for this function.